### DISCUSSION TITLE
In place matrix/tensor update

### DISCUSSION THREAD

**Discussion by TuanNguyen27** [jax-ml/jax#4492] [Q&A] [answered]
I know that JAX explicitly doesn't support in place operation, but i saw this argument about memory location in `Dive Into Deep Learning` book. I'm wondering if there's a way to achieve this in JAX, e.g update a tensor and keep the same memory location. 

 
<img width="930" alt="Screen Shot 2020-10-07 at 9 36 07 PM" src="https://user-images.githubusercontent.com/13680821/95415923-641c2b00-08e6-11eb-9c3b-b699d153a9ac.png">

The standard JAX way to do the above code would be (i think)
```
Z = jnp.zeros_like(Y)
print('id(Z):', id(Z))
Z = jax.ops.index_update(Z, jax.ops.index[:], X+Y)
print('id(Z):', id(Z))
```
but as you can tell, it does not retain the original memory location
```
id(Z): 140308028317280
id(Z): 140308028316752
```

Would love to hear more about this! Is the issue about memory reference being stale valid, and how do JAX frameworks tackle this ?

**Comment by jakevdp** [comment, marked as answer]
See https://github.com/google/jax/issues/1733 for a version of this request and some recent updates.
